MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CustomerServiceDefault.java

```diff
@@ -51,26 +53,26 @@
 
     final Scheduler callbackScheduler;
 
-    CustomerToken customerToken;
+    final AtomicReference<CustomerToken> customerTokenRef = new AtomicReference<>();
 
     CustomerServiceDefault(
             final Retrofit retrofit,
             final CustomerToken customerToken,
             final Scheduler callbackScheduler
     ) {
         this.retrofitService = retrofit.create(CustomerRetrofitService.class);
-        this.customerToken = customerToken;
+        this.customerTokenRef.set(customerToken);
         this.callbackScheduler = callbackScheduler;
     }
 
     @Override
-    public void setCustomerToken(CustomerToken customerToken) {
-        this.customerToken = customerToken;
-    }
-
-    @Override
     public CustomerToken getCustomerToken() {
-        return customerToken;
+        final CustomerToken customerToken = customerTokenRef.get();
+        if (customerToken != null) {
+            return new CustomerToken(customerToken);
```

Comment:
From best practices, you should restrict the mutation outside of class. For instance if someone decides to change inner state of this object that means it will affect and the most dangerous implicitly the BuyClient.